woocommerce/woocommerce-ios
# Bulk Order Actions: Add Swipe Actions Glance Animation

Closes: #7343 

# Why

After "Swipe to Complete" is fully implemented, it is time to inform the users that the feature exists. The way we are doing it is by adding a short glance animation to reveal the underlying swipe actions of a given order.

As there is no way to properly glance the swipe actions of a cell, this PR makes a dirty trick to simulate it. The animation does the following:

- Identifies the first cell that has a swipe action.
- Animates the cell out.
- Place a background underneath the cell with the same swipe action background color.
- Animates the cell in.
- Remove the temporary background

Additionally, the animation follows these rules:

- Only glance the first visible cell that contains a swipe action after the results are loaded.
- Only perform the glance animation once per app lifecycle.

# How

This first change comes in the **UITableView+Helpers** extension, There I'm adding a `glanceTrailingSwipeActions()` method that searches for swipe actions in visible cells, if any action is found, then the glance animation is performed.

The second change comes in the **OrderListViewController** class, where I'm adding a control variable to know if the glance animation needs to be performed or not.

# Demo

https://user-images.githubusercontent.com/562080/181792646-41a69ce5-136c-4b61-8342-02f1e8c9ed8c.mov

# Testing Steps

- Make sure your store has a non-completed order on the first page.
- Open the app & navigate to the order list
- See the glance animation being performed.

---
